File: app/src/main/java/com/example/kocja/rabbiter_reworked/services/alertIfNotAlertedService.java
package com.example.kocja.rabbiter_reworked.services;

import android.app.AlarmManager;
import android.app.IntentService;
import android.app.PendingIntent;
import android.content.Context;
import android.content.Intent;
import android.support.annotation.Nullable;
import android.util.Log;

import com.example.kocja.rabbiter_reworked.GsonManager;
import com.example.kocja.rabbiter_reworked.HttpManager;
import com.example.kocja.rabbiter_reworked.broadcastrecievers.NotifReciever;
import com.example.kocja.rabbiter_reworked.databases.Events;
import com.google.gson.JsonObject;

import okhttp3.Response;

/**
 * Created by kocja on 28/02/2018.
 */

public class alertIfNotAlertedService extends IntentService {
    public alertIfNotAlertedService(){
        super("This is alertIfNotAlertedService");
    }
    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        HttpManager.getRequest("seekNotAlertedEvents", response -> {
            Intent startNotificationIntent = new Intent(alertIfNotAlertedService.this, NotifReciever.class);
            AlarmManager manager = (AlarmManager)getSystemService(Context.ALARM_SERVICE);
            for(Events event : GsonManager.getGson().fromJson(response,Events[].class)){
                Log.v("Oops","This guy was not started");
                startNotificationIntent.putExtra("eventUUID",event.eventUUID);
                PendingIntent startNotification = PendingIntent.getBroadcast(alertIfNotAlertedService.this,event.id,startNotificationIntent,PendingIntent.FLAG_UPDATE_CURRENT);
                //manager.set(AlarmManager.RTC_WAKEUP,event.dateOfEvent.getTime(),startNotification);
            }
        });
        /*
        SQLite.select()
                .from(Events.class)
                .where(Events_Table.notificationState.eq(Events.NOT_YET_ALERTED))
                .async()
                .queryListResultCallback((transaction, tResult) -> {
                }).execute();
          */
    }
}

File: app/src/main/java/com/example/kocja/rabbiter_reworked/services/alertIfNotAlertedService.kt
package com.example.kocja.rabbiter_reworked.services

import android.app.AlarmManager
import android.app.IntentService
import android.app.PendingIntent
import android.content.Context
import android.content.Intent
import android.util.Log

import com.example.kocja.rabbiter_reworked.broadcastrecievers.NotifReciever
import com.example.kocja.rabbiter_reworked.databases.Events
import com.example.kocja.rabbiter_reworked.databases.Events_Table
import com.raizlabs.android.dbflow.sql.language.SQLite

/**
 * Created by kocja on 28/02/2018.
 */

class alertIfNotAlertedService : IntentService("This is alertIfNotAlertedService") {
    override fun onHandleIntent(intent: Intent?) {

        SQLite.select()
                .from(Events::class.java)
                .where(Events_Table.notificationState.eq(Events.NOT_YET_ALERTED))
                .async()
                .queryListResultCallback { _, tResult ->
                    val startNotificationIntent = Intent(this, NotifReciever::class.java)
                    val manager = getSystemService(Context.ALARM_SERVICE) as AlarmManager
                    for (event in tResult) {
                        Log.v("Oops", "This guy was not started")
                        startNotificationIntent.putExtra("eventUUID", event.eventUUID)
                        val startNotification = PendingIntent.getBroadcast(this, event.id, startNotificationIntent, PendingIntent.FLAG_UPDATE_CURRENT)
                        manager.set(AlarmManager.RTC_WAKEUP, event.dateOfEvent!!.time, startNotification)
                    }
                }.execute()
    }
}
